### PR TITLE
Restructure GenericMessageEvent

### PR DESCRIPTION
**Pull request by WalshyDev**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

There are several guidelines you should follow in order for your
  Pull Request to be merged.

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

> It is sometimes better to include more changes in a single commit. 
  If you find yourself having an overwhelming amount of commits, you
  can **rebase** your branch.

## Description
This is a simple PR, this just adds `getGuild()` to GenericMessageEvent since it is just added in implementations.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/core/events/message/MessageBulkDeleteEvent.java [hunk 1]
```diff
@@ -46,11 +46,6 @@ public TextChannel getChannel()
         return channel;
     }
 
-    public Guild getGuild()
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/core/events/message/MessageBulkDeleteEvent.java]: Do you realize that this event doesn't extend the GenericMessageEvent?

**Comment by MinnDevelopment** [comment]
While you are at it, why not also move all the getXChannel methods up?

File: src/main/java/net/dv8tion/jda/core/events/message/MessageBulkDeleteEvent.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/net/dv8tion/jda/core/events/message/MessageBulkDeleteEvent.java]: Oh whoops, reverted

File: src/main/java/net/dv8tion/jda/core/events/message/GenericMessageEvent.java [hunk 2]
```diff
@@ -64,4 +68,59 @@ public ChannelType getChannelType()
...
+     * <br>If this Message was not received in a {@link net.dv8tion.jda.core.entities.TextChannel TextChannel},
+     * this will return {@code null}.
+     *
+     * @return The Guild the Message was received in or null if not from a TextChannel
+     *
+     * @see net.dv8tion.jda.core.events.message.GenericMessageEvent#isFromType(ChannelType)
+     */
+    public Guild getGuild()
+    {
+        return isFromType(ChannelType.TEXT) ? ((TextChannel) channel).getGuild() : null;
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/core/events/message/GenericMessageEvent.java]: change this to getTextChannel().getGuild()

**Pull request merged**